firefox-ios/Client/Frontend/Browser/SearchEngines/Redux/SearchEngineSelectionMiddleware.swift

```diff
@@ -28,7 +29,18 @@ final class SearchEngineSelectionMiddleware {
             guard !searchEngines.isEmpty else {
                 // The SearchEngineManager should have loaded these by now, but if not, attempt to fetch the search engines
                 self.searchEnginesManager.getOrderedEngines { [weak self] preferences, searchEngines in
-                    self?.notifyDidLoad(windowUUID: action.windowUUID, searchEngines: searchEngines)
+                    // TODO: FXIOS-12943 Once we are on Swift 6, we can mark this closure @MainActor,
+                    // since it is dispatched to the main queue behind the scenes under DefaultSearchEngineProvider
+                    guard Thread.isMainThread else {
+                        self?.logger.log("Completion is not called on the main thread",
+                                         level: .fatal,
+                                         category: .storage)
+                        return
+                    }
+
+                    MainActor.assumeIsolated {
+                        self?.notifyDidLoad(windowUUID: action.windowUUID, searchEngines: searchEngines)
```

Comment:
We get this error otherwise:

<img width="1404" height="632" alt="Screenshot 2025-07-24 at 4 05 56 PM" src="https://github.com/user-attachments/assets/2a7e4e35-4c73-4b39-b523-d99c6bd8d6f9" />